### PR TITLE
Tests the new HTTP proxy configuration on a real envrionment

### PR DESCRIPTION
**Pull request by shadab-geo**
- Update the way in which HttpRoutePlanner is set for HttpClientBuilder

### CODE REVIEW

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java [hunk 1]
```diff
@@ -156,30 +156,36 @@ public HttpClient createHttpClient() {
...
-                        final HttpRequest request,
-                        final HttpContext context) throws HttpException {
-                    String hostname = host.getHostName();
-                    if (isNonProxyHost(hostname)) {
-                        // Return direct route
-                        return new HttpRoute(host);
+        final HttpHost httpHost = getHost("http.proxyHost", "http.proxyPort");
+        LOGGER.debug("http host: " + httpHost);
+        final HttpHost httpsHost = getHost("https.proxyHost", "https.proxyPort");
+        LOGGER.debug("https host: " + httpsHost);
```
Comment (nmco) [inline src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:162]: As I say above the HTTP proxy business should be isolated in a method, the log should (and variables) should be more meaningful. 

Imagine someone reading the log and he seems something like this:
```
https host: outputvm.net25
```
Unless you wrote, this means absolutely zero.

While if we have a log like (DEBUG LEVEL):
```
HTTP proxy host: outputvm.net25
HTTPS proxy host: NOT DEFINED
```

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java [hunk 2]
```diff
@@ -156,30 +156,36 @@ public HttpClient createHttpClient() {
...
+
+        HttpRoutePlanner routePlanner = new HttpRoutePlanner() {
+            public HttpRoute determineRoute(
+                    HttpHost target,
+                    HttpRequest request,
+                    HttpContext context) {
+                LOGGER.info("target host: " + target);
+                if (isNonProxyHost(target.getHostName())) {
```
Comment (nmco) [inline src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:170]: At the INFO level we should be able to understand which requests the HTTP proxy received and how they were routed. At debug, we should understand the inner workflow and context (fr example like above we look at the proxy configuration).

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java [hunk 3]
```diff
@@ -209,13 +215,15 @@ private Boolean isNonProxyHost(String host) {
             if (nonProxyHostProp.endsWith("\"")) {
                 nonProxyHostProp = nonProxyHostProp.substring(0, nonProxyHostProp.length() - 1);
             }
+            LOGGER.info("http.nonProxyHosts: " + nonProxyHostProp);
             StringTokenizer tokenizer = new StringTokenizer(nonProxyHostProp, "|");
             while (tokenizer.hasMoreTokens()) {
                 String str = tokenizer.nextToken().trim();
-                str = str.replace("*", "\\w*");
+                str = str.replace("*", "[\\w-]*");
```
Comment (nmco) [inline src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:222]: The parsing of non should only be done once, do you confirm is the case?

Secondly, is the comparison case sensitive? Please double check with Oracle spec.

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java [hunk 4]
```diff
@@ -156,30 +156,36 @@ public HttpClient createHttpClient() {
...
-                @Override
-                public HttpRoute determineRoute(
-                        final HttpHost host,
-                        final HttpRequest request,
-                        final HttpContext context) throws HttpException {
-                    String hostname = host.getHostName();
-                    if (isNonProxyHost(hostname)) {
-                        // Return direct route
-                        return new HttpRoute(host);
+        final HttpHost httpHost = getHost("http.proxyHost", "http.proxyPort");
```
[the remark this answers is not in the document]
Comment (shadab-geo) [reply src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:159]: updated

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java (hunk 1 above)
Comment (shadab-geo) [reply src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:162]: updated the logs

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java (hunk 3 above)
Comment (shadab-geo) [reply src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:222]: 1. I could see that this parsing is done for every request.
2. There is no explicit mention of case sensitivity on oracle specs. It refers to matching the ```patterns ``` which means that the comparison is case sensitive.

File: src/main/java/it/geosolutions/httpproxy/HTTPProxy.java [hunk 5]
```diff
@@ -199,6 +187,40 @@ HttpHost getHost(String proxyHostKey, String proxyPortKey) {
         return httpHost;
     }
 
+    /**
+     * Returns the HttpRoutePlanner based on the target host http scheme
+     * @param httpHost
+     * @param httpsHost
+     * @return HttpRoutePlanner
+     */
```
Comment (nmco) [inline src/main/java/it/geosolutions/httpproxy/HTTPProxy.java:195]: These parameters Javadoc don't tell us anything useful, they are just noise, please remove them.

This Javadoc doesn't tell us much:
```
Returns the HttpRoutePlanner based on the target host http scheme
```

**Comment by nmco** [comment]
Thank you for applying the changes @shadab-geo, that says:
- I'm not 100% comfortable with the changes we are pushing, for example, the way we are setting up the proxy host looks too clumsy to me, why not use the method described [HERE](https://hc.apache.org/httpcomponents-client-4.5.x/current/tutorial/html/connmgmt.html) using default proxy planner?
- Doing these things without integration tests or manual tests is less than ideal, we need to find a way to test this, maybe using docker?

To avoid blocking everyone, I'm going to merge this, but I have created [THIS](https://github.com/geosolutions-it/mapstore2-c027/issues/97) issue so we can fix things properly.

**Pull request merged**